lib/pbench/server/api/resources/pbench_users.py

```diff
@@ -0,0 +1,426 @@
+            self.logger.warning("Invalid email {}", emailID)
+            abort(400, message=f"Invalid email: {emailID}")
+
+        # check if user already exist
+        try:
+            user = User.query(username=user_data.get("username"))
+        except Exception:
+            self.logger.exception("Exception while querying user")
+            abort(500, message="INTERNAL ERROR")
+
+        if user:
+            self.logger.warning(
+                "A user tried to re-register. Username: {}", user.username
+            )
+            abort(403, message="A user with that name already exists.")
+
+        try:
+            user = User(
+                bcrypt_log_rounds=self.bcrypt_log_rounds,
+                username=username,
+                password=password,
+                firstName=firstName,
+                lastName=lastName,
+                email=email,
+            )
+
+            # insert the user
+            user.add()
+            self.logger.info(
+                "New user registered, username: {}, email: {}", username, email
+            )
+
+            response_object = {
+                "status": "success",
+                "message": "Successfully registered.",
+            }
+            response = jsonify(response_object)
+            response.status_code = 201
+            return make_response(response, 201)
+        except Exception:
+            self.logger.exception("Exception while registering a user")
+            abort(500, message="INTERNAL ERROR")
+
+
+class LoginAPI(Resource):
+    """
+    Pbench API for User Login or generating an auth token
+    """
+
+    def __init__(self, config, logger, auth):
+        self.server_config = config
+        self.logger = logger
+        self.auth = auth
+        try:
+            self.token_expire_duration = self.server_config.get(
+                "pbench-server", "token_expiration_duration"
+            )
+        except NoOptionError:
+            self.token_expire_duration = None
+
+    @Auth.token_auth.login_required(optional=True, f=Auth().verify_auth())
+    def post(self):
+        """
+        Post request for logging in user.
+        The user is allowed to re-login multiple times and each time a new valid auth token will be provided
+
+        This requires a JSON data with required user metadata fields
+        {
+            "username": "username",
+            "password": "password",
+        }
+
+        Required headers include
+
+            Content-Type:   application/json
+            Accept:         application/json
+
+        :return: JSON Payload
+        if we succeed to decrypt the password hash, the returned response_object will include the auth_token
+            response_object = {
+                    "status": "success", # will not present if failed
+                    "message": "Successfully logged in."/"failure message",
+                    "auth_token": auth_token.decode(), # Will not present if failed
```

Comment:
You're returning `auth_token`, not `auth_token.decode()` ... and I think that's what we want. However, for the docstring, something that's less dependent on understanding the  code would be more useful, like maybe just `"auth_token": "<authorization_token>"`

Reply:
Fixed, also with newer pyjwt 2.0.0 package encoding auth token returns a string instead of bytes to so need to decode it.